libcontainer/cgroups/devices/systemd.go

```diff
@@ -133,8 +133,18 @@ func systemdProperties(r *configs.Resources) ([]systemdDbus.Property, error) {
 			// rules separately to systemd) we can safely skip entries that don't
 			// have a corresponding path.
 			if _, err := os.Stat(entry.Path); err != nil {
-				logrus.Debugf("skipping device %s for systemd: %s", entry.Path, err)
-				continue
+				// Also check /sys/dev so that we don't depend on /dev/{block,char}
+				// being populated. (/dev/{block,char} is populated by udev, which
+				// isn't strictly required for systemd). Ironically, this happens most
+				// easily when starting containerd within a runc created container
+				// itself.
+
+				// We don't bother with securejoin here because we create entry.Path
+				// right above here, so we know it's safe.
+				if _, err := os.Stat("/sys" + entry.Path); err != nil {
```

Comment:
I just realised we probably want to do `entry.Path = "/sys" + entry.Path` here, don't we? Otherwise systemd will get a `/dev/...` path that doesn't exist?

Reply:
So I’m not entirely clear on what systemd does with these values but I can tell you that in my setup, those /dev/char entries don’t exist and systemd still does the right thing. Maybe it parses the path and uses the major:minor directly?

Reply:
Nah, the `/sys/dev/{block,char}/<major>:<minor>` is a directory, not a device file, so we can't use it for systemd.

Perhaps systemd has a set of built-in devices it can recognize. I'll take a look.